crates/bevy_asset/src/asset_server.rs

```diff
@@ -242,37 +247,75 @@ impl AssetServer {
             .cloned()
     }
 
-    /// Gets the load state of an asset from the provided handle.
-    pub fn get_load_state<H: Into<HandleId>>(&self, handle: H) -> LoadState {
+    /// Gets the source info of an asset from the provided handle.
+    pub fn get_source_info<H: Into<HandleId>>(
+        &self,
+        handle: H,
+    ) -> Option<impl Deref<Target = SourceInfo> + '_> {
```

Comment:
Why can't we just return a `Option<SourceInfo>` here?

Reply:
I didn't want to clone it, as it's not entirely cheap to do so, but I agree that this interface kind-of sucks :sweat_smile: We can't have a plain reference because the data's behind a lock, so this is the next best thing.

If you think cloning is better, I'll fix that quickly, and change the implementation of methods that rely on `get_source_info` to access the field `source_info` directly instead.

Reply:
Hmm. I think that's fine, just leave a comment explaining this in the docs for that method.

Reply:
Sorry, I got busy with work this past month and let this unresolved.  I just added the missing comment.